docs/reference.md

```diff
@@ -374,6 +376,12 @@ Returns the the current account ID (“Hub ID” or “portal ID”) for the pag
 
 Returns `true` for components rendered live for a deployed project and `false` when rendering in the dev server.
 
+### `getSecret`
+
+`(secretName: string) => string`
+
+Returns a value for a given secret key. The secret must be defined using `hs secrets` in the CLI and the key must be included in a `secrets` array in your `cms-assets.json` configuration. When working  To prevent accidental leaking of secrets, `getSecret()` will only successfully run on the server and not within an island. If a secret value isn't sensitive and you need to access it in island components, you may call `getSecret()` outside the island and pass the value in as a prop.
```

Comment:
```suggestion
Returns a value for a given secret key. The secret must be defined using `hs secrets` in the CLI and the key must be included in a `secrets` array in your `cms-assets.json` configuration. In order to prevent accidentally leaking  secrets, `getSecret()` can only be called from components executed on the server and not from the browser (within an island). If a secret value isn't sensitive and you need to access it in island components, you may call `getSecret()` outside the island and pass the value down via a prop.
```